src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -39,7 +44,8 @@ public List<User> createUsers(final List<String> emails) {
      * @return map {user email : user}
      */
     public Map<String, User> groupByEmail(final List<User> users) {
-        return null;
+        return users.stream()
+                .collect(Collectors.toMap(User::getEmail, user -> user, (a, b) -> b));
```

Comment:
Providing merging function is good if you want to allow duplicates but in this particular case it's really unnecessary.